container/flv/encoder.go

```diff
@@ -166,93 +169,147 @@ func (s *frameScanner) readByte() (b byte, ok bool) {
 		// section E.3.
 		var zero [4]byte
-		_, err := e.dst.Write(zero[:])
+		written, err := e.dst.Write(zero[:])
+		totalWritten += written
 		if err != nil {
-			return 0, err
+			return totalWritten, err
 		}
 	}
 	timeStamp := e.getNextTimestamp()
-	// Do we have video to send off?
-	if e.video {
-		if isKeyFrame(frame) {
-			frameType = KeyFrameType
-		} else {
-			frameType = InterFrameType
-		}
-		if isSequenceHeader(frame) {
-			packetType = SequenceHeader
-		} else {
-			packetType = AVCNALU
-		}
+	if isKeyFrame(videoFrame) {
+		frameType = KeyFrameType
+	} else {
+		frameType = InterFrameType
+	}
+	if isSequenceHeader(videoFrame) {
+		packetType = SequenceHeader
+	} else {
+		packetType = AVCNALU
+	}
 
-		tag := VideoTag{
-			TagType:           uint8(VideoTagType),
-			DataSize:          uint32(len(frame)) + DataHeaderLength,
-			Timestamp:         timeStamp,
-			TimestampExtended: NoTimestampExtension,
-			FrameType:         frameType,
-			Codec:             H264,
-			PacketType:        packetType,
-			CompositionTime:   0,
-			Data:              frame,
-			PrevTagSize:       uint32(videoHeaderSize + len(frame)),
-		}
-		_, err := e.dst.Write(tag.Bytes())
-		if err != nil {
-			return len(frame), err
-		}
+	tag := VideoTag{
+		TagType:           uint8(VideoTagType),
+		DataSize:          uint32(len(videoFrame)) + DataHeaderLength,
+		Timestamp:         timeStamp,
+		TimestampExtended: NoTimestampExtension,
+		FrameType:         frameType,
+		Codec:             H264,
+		PacketType:        packetType,
+		CompositionTime:   0,
+		Data:              videoFrame,
+		PrevTagSize:       uint32(videoHeaderSize + len(videoFrame)),
+	}
+	written, err := e.dst.Write(tag.Bytes())
+	totalWritten += written
+	if err != nil {
+		return totalWritten, err
 	}
-	// Do we even have some audio to send off ?
-	if e.audio {
-		// Not sure why but we need two audio tags for dummy silent audio
-		// TODO: create constants or SoundSize and SoundType parameters
-		tag := AudioTag{
-			TagType:           uint8(AudioTagType),
-			DataSize:          7,
-			Timestamp:         timeStamp,
-			TimestampExtended: NoTimestampExtension,
-			SoundFormat:       AACAudioFormat,
-			SoundRate:         3,
-			SoundSize:         true,
-			SoundType:         true,
-			Data:              dummyAudioTag1Data,
-			PrevTagSize:       uint32(audioSize),
-		}
-		_, err := e.dst.Write(tag.Bytes())
-		if err != nil {
-			return len(frame), err
-		}
 
-		tag = AudioTag{
-			TagType:           uint8(AudioTagType),
-			DataSize:          21,
-			Timestamp:         timeStamp,
-			TimestampExtended: NoTimestampExtension,
-			SoundFormat:       AACAudioFormat,
-			SoundRate:         3,
-			SoundSize:         true,
-			SoundType:         true,
-			Data:              dummyAudioTag2Data,
-			PrevTagSize:       uint32(22),
-		}
-		_, err = e.dst.Write(tag.Bytes())
+	return totalWritten, nil
+}
+
+// An adapter that provides the io.Writer interface for WriteVideo.
+type VideoWriterAdapter struct {
```

Comment:
I'm not sure I understand why we need this. Can't we just leave the Write method as video ?

Reply:
The minimal change would be to just expose a method on the encoder to modify the default behaviour though ? At this stage we're really just testing this as a proof of concept. There's a good chance there'll be a refactor that will make everything nicer here anyways, and it probably won't look like what you've done, so we can just deal with this then.

Reply:
In general it's a good practice to avoid interface modifications if you can help it. Additions are preferred to modifications on existing interfaces if you're just looking to add some knobs for behaviour modification.